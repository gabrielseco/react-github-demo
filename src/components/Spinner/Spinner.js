// @flow
import React from 'react';
import styles from './Spinner.scss';

const Spinner = () => (
  <div
    className={`${styles.spinner} spinner-test`}
    role="alertdialog"
    aria-busy="true"
    aria-live="assertive"
  />
);

export default Spinner;
