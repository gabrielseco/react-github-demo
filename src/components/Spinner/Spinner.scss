// Props go to https://loading.io/css/

$spinner-color: #212240;
$spinner-width: 64px;

.spinner {
  display: inline-block;
  height: $spinner-width;
  width: $spinner-width;
}

.spinner::after {
  animation: lds-dual-ring 1.2s linear infinite;
  border: 5px solid $spinner-color;
  border-color: $spinner-color transparent $spinner-color transparent;
  border-radius: 50%;
  content: '';
  display: block;
  height: 46px;
  margin: 1px;
  width: 46px;
}

@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
