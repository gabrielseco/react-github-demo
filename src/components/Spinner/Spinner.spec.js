import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Spinner from './Spinner';

describe('Spinner suite', () => {
  it('should render the Spinner component', () => {
    const component = shallow(<Spinner />);
    const tree = renderer.create(<Spinner />).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
