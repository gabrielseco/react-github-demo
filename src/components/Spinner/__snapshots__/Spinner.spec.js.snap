// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner suite should render the Spinner component 1`] = `
<div
  aria-busy="true"
  aria-live="assertive"
  className="spinner spinner-test"
  role="alertdialog"
/>
`;
